airbyte-integrations/connectors/destination-snowflake/src/main/resources/spec.json

```diff
@@ -240,6 +247,83 @@
                 "description": "Whether to delete the staging files from S3 after completing the sync. See the docs for details. Only relevant for COPY. Defaults to true.",
                 "default": true,
                 "order": 6
+              },
+              "encryption": {
+                "title": "Encryption",
+                "type": "object",
+                "description": "How to encrypt the staging data",
+                "default": { "encryption_type": "none" },
+                "order": 7,
+                "oneOf": [
+                  {
+                    "title": "No encryption",
+                    "description": "Staging data will be stored in plaintext.",
+                    "type": "object",
+                    "required": ["encryption_type"],
+                    "properties": {
+                      "encryption_type": {
+                        "type": "string",
+                        "const": "none",
+                        "enum": ["none"],
+                        "default": "none"
+                      }
+                    }
+                  },
+                  {
+                    "title": "AES-CBC envelope encryption",
+                    "description": "Staging data will be encrypted using AES-CBC envelope encryption.",
+                    "type": "object",
+                    "required": ["encryption_type", "key_encrypting_key"],
+                    "properties": {
+                      "encryption_type": {
+                        "type": "string",
+                        "const": "aes_cbc_envelope",
+                        "enum": ["aes_cbc_envelope"],
+                        "default": "aes_cbc_envelope"
+                      },
+                      "key_encrypting_key": {
+                        "title": "Key-encrypting key",
+                        "description": "The type of key-encrypting key to use",
+                        "type": "object",
+                        "oneOf": [
```

Comment:
You should test the spec in storybook to see if it works. Previously our UI does not support multiple-level `oneOf` specs very well.

Reply:
This can be replaced with a simple text field. When it is not empty, it is a user provided key. When it is empty, it is an ephemeral key.

Reply:
I think nested oneOf work well now (tested in storybook, and also with a local airbyte instance). There's some weirdness with empty objects, but that seems like an actual bug which is also breaking e.g. the Internal Staging option - https://github.com/airbytehq/airbyte/issues/12457

but will switch to the text field anyway, it would make the spec simpler to read + parse